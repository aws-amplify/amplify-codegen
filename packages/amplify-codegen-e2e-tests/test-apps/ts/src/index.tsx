import React from 'react';
import ReactDOM from 'react-dom/client';
import { Amplify } from 'aws-amplify';
import App from './App';
import awsconfig from './aws-exports';

Amplify.configure(awsconfig);
const root = ReactDOM.createRoot(document.getElementById('root') as HTMLElement);
root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
);
