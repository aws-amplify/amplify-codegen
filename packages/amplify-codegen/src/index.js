const generateTypes = require('./commands/types');
const generateStatements = require('./commands/statements');
const generate = require('./commands/generateStatementsAndType');
const add = require('./commands/add');
const remove = require('./commands/remove');
const configure = require('./commands/configure');
const generateModels = require('./commands/models');
const { generateModelIntrospection, getModelIntrospection } = require('./commands/model-intropection');
const { isCodegenConfigured, switchToSDLSchema } = require('./utils');
const prePushAddGraphQLCodegenHook = require('./callbacks/prePushAddCallback');
const prePushUpdateGraphQLCodegenHook = require('./callbacks/prePushUpdateCallback');
const postPushGraphQLCodegenHook = require('./callbacks/postPushCallback');
const { executeAmplifyCommand, handleAmplifyEvent } = require('./amplify-plugin-index');
const { getCodegenConfig } = require('./codegen-config');

module.exports = {
  configure,
  generate,
  generateTypes,
  generateStatements,
  add,
  remove,
  prePushAddGraphQLCodegenHook,
  prePushUpdateGraphQLCodegenHook,
  postPushGraphQLCodegenHook,
  isCodegenConfigured,
  switchToSDLSchema,
  executeAmplifyCommand,
  handleAmplifyEvent,
  generateModels,
  generateModelIntrospection,
  getModelIntrospection,
  getCodegenConfig,
};

// temp change to trigger release
