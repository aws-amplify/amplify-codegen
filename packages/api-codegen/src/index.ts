export type { GenerateModelsOptions } from './models';
export { generateModels } from './models';
