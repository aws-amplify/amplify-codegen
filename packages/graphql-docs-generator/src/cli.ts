import * as yargs from 'yargs';
import { logError } from './logger';
import { generateGraphQLDocuments } from './index';

// Make sure unhandled errors in async code are propagated correctly
process.on('unhandledRejection', error => {
  throw error;
});

process.on('uncaughtException', handleError);

function handleError(error: Error) {
  logError(error);
  process.exit(1);
}

export function run(argv: Array<String>): void {
  // tslint:disable
  yargs
    .command(
      '$0',
      'Generate graphql operations for the provided introspection schema',
      {
        schema: {
          demand: true,
          describe: 'GraphQL introspection or SDL schema',
          normalize: true,
        },
        maxDepth: {
          demand: true,
          default: 2,
          normalize: true,
          type: 'number',
        },
        typenameIntrospection: {
          default: true,
          type: 'boolean',
        },
      },
      async argv => {
        generateGraphQLDocuments(argv.schema, { maxDepth: argv.maxDepth, typenameIntrospection: argv.typenameIntrospection });
      }
    )
    .help()
    .version()
    .strict().argv;
}
