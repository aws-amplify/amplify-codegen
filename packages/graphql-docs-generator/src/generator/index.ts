export * from './types';
import generate from './generate';
export { generateMutations, generateSubscriptions, generateQueries, lowerCaseFirstLetter } from './generateAllOperations';
export { buildSchema } from './utils/loading';
export { getTemplatePartials, getOperationPartial, getExternalFragmentPartial } from './utils/templates';
export default generate;
