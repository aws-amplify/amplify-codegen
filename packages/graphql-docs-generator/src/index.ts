const handlebars = require('handlebars/dist/handlebars');
import generateAllOps, { GQLTemplateOp, GQLAllOperations, GQLTemplateFragment, lowerCaseFirstLetter } from './generator';
import { buildSchema } from './generator/utils/loading';
import { getTemplatePartials, getOperationPartial, getExternalFragmentPartial } from './generator/utils/templates';
export { buildSchema } from './generator/utils/loading';

export function generateGraphQLDocuments(
  schema: string,
  options: { maxDepth?: number, useExternalFragmentForS3Object?: boolean; typenameIntrospection?: boolean },
): GeneratedOperations {
  const opts = {
    maxDepth: 2,
    useExternalFragmentForS3Object: true,
    typenameIntrospection: true,
    ...options,
  };

  const extendedSchema = buildSchema(schema);

  const gqlOperations: GQLAllOperations = generateAllOps(extendedSchema, opts.maxDepth, {
    useExternalFragmentForS3Object: opts.useExternalFragmentForS3Object,
    typenameIntrospection: opts.typenameIntrospection
  });
  registerPartials();
  registerHelpers();

  const allOperations = {
    queries: new Map<string, string>(),
    mutations: new Map<string, string>(),
    subscriptions: new Map<string, string>(),
    fragments: new Map<string, string>()
  };

  ['queries', 'mutations', 'subscriptions'].forEach(op => {
    const ops = gqlOperations[op];
    if (ops.length) {
      const renderedOperations = renderOperations(gqlOperations[op]);
      allOperations[op] = renderedOperations;
    }
  });

  if (gqlOperations.fragments.length) {
    const renderedFragments = renderFragments(gqlOperations.fragments, opts.useExternalFragmentForS3Object);
    allOperations['fragments'] = renderedFragments;
  }

  return allOperations;
}

type GeneratedOperations = {
  queries: Map<string, string>;
  mutations: Map<string, string>;
  subscriptions: Map<string, string>;
  fragments: Map<string, string>;
}

function renderOperations(operations: Array<GQLTemplateOp>): Map<string, string> {
  const renderedOperations = new Map<string, string>();
  if (operations?.length) {
    operations.forEach(op => {
      const name = op.fieldName || op.name;
      const gql = renderOperation(op);
      renderedOperations.set(name, gql);
    });
  }

  return renderedOperations;
}

function renderOperation(operation: GQLTemplateOp): string {
  const templateStr = getOperationPartial();
  const template = handlebars.compile(templateStr, {
    noEscape: true,
    preventIndent: true,
  });
  return template(operation);
}

function renderFragments(fragments: Array<GQLTemplateFragment>, useExternalFragmentForS3Object: boolean): Map<string, string> {
  const renderedFragments = new Map<string, string>();
  if (fragments?.length) {
    fragments.forEach(fragment => {
      const name = fragment.name;
      const gql = renderFragment(fragment,useExternalFragmentForS3Object );
      renderedFragments.set(name, gql);
    });
  }

  return renderedFragments;
}

function renderFragment(fragment: GQLTemplateFragment, useExternalFragmentForS3Object: boolean): string {
  if (!useExternalFragmentForS3Object) {
    return;
  }

  const templateStr = getExternalFragmentPartial();
  const template = handlebars.compile(templateStr, {
    noEscape: true,
    preventIndent: true,
  });
  return template(fragment);
}

function registerPartials() {
  const partials = getTemplatePartials();
  for (const [partialName, partialContent] of Object.entries(partials)) {
    handlebars.registerPartial(partialName, partialContent);
  }
}

function registerHelpers() {
  const formatNameHelper = lowerCaseFirstLetter;
  handlebars.registerHelper('formatName', formatNameHelper);
}
