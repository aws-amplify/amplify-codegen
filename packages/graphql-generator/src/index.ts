export { generateModels, generateModelsSync } from './models';
export { generateStatements } from './statements';
export { generateTypes } from './types';

export type {
  TypesTarget,
  ModelsTarget,
  StatementsTarget,
  GenerateTypesOptions,
  GenerateModelsOptions,
  GenerateStatementsOptions,
  GeneratedOutput
} from './typescript';
