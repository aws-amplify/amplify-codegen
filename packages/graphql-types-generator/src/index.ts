import './polyfills';

export { default as generate, generateFromString } from './generate';
export type { Target } from './types';
export { getOutputFileName } from './utilities/getOutputFileName';
export { extractDocumentFromJavascript } from './loading';
