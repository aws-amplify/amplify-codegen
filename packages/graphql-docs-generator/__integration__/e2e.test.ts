import { resolve } from 'path';
import * as fs from 'fs';
import { generateGraphQLDocuments } from '../src';

describe('GraphQL documents generation tests for introspection schema input', () => {
  const schemaPath = resolve(__dirname, '../fixtures/schema.json');
  const schema = fs.readFileSync(schemaPath, 'utf8');

  it('should generate GraphQL documents for schema', () => {
    const generatedOutput = generateGraphQLDocuments(schema, { maxDepth: 3 });
    snapshotTestGeneratedOutput(generatedOutput);
  });
});

describe('end 2 end tests to test if the case style is retained for type names', () => {
  const schemaPath = resolve(__dirname, '../fixtures/caseTypes.graphql');
  const schema = fs.readFileSync(schemaPath, 'utf8');

  it('should generate statements', () => {
    const generatedOutput = generateGraphQLDocuments(schema, { maxDepth: 3 });
    snapshotTestGeneratedOutput(generatedOutput);
  });
});

const snapshotTestGeneratedOutput = (generatedOutput: any, includesFragments = false) => {
  expect(generatedOutput.queries).toBeDefined();
  expect(generatedOutput.queries).toMatchSnapshot();

  expect(generatedOutput.mutations).toBeDefined();
  expect(generatedOutput.mutations).toMatchSnapshot();

  expect(generatedOutput.subscriptions).toBeDefined();
  expect(generatedOutput.subscriptions).toMatchSnapshot();

  if (includesFragments) {
    expect(generatedOutput.fragments).toBeDefined();
    expect(generatedOutput.fragments).toMatchSnapshot();
  }
};
