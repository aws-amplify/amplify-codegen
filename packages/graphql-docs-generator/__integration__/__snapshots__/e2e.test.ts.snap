// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphQL documents generation tests for introspection schema input should generate GraphQL documents for schema 1`] = `
Map {
  "hero" => "
query Hero 
  (
      $episode:Episode
  )
{
  hero
  (
      episode:$episode
  )
 
   {
    
    id
    name
      friends {
        
    id
    name
      friends {
        
    id
    name
    appearsIn
        
    ...on Human {
      
    homePlanet
    height
    mass
    }
    ...on Droid {
      
    primaryFunction
    }
      }
      friendsConnection {
        
    totalCount
    __typename
        
      }
    appearsIn
        
    ...on Human {
      
    homePlanet
    height
    mass
      starships {
        
    id
    name
    length
    coordinates
    __typename
        
      }
    }
    ...on Droid {
      
    primaryFunction
    }
      }
      friendsConnection {
        
    totalCount
      edges {
        
    cursor
    __typename
        
      }
      friends {
        
    id
    name
    appearsIn
        
    ...on Human {
      
    homePlanet
    height
    mass
    }
    ...on Droid {
      
    primaryFunction
    }
      }
      pageInfo {
        
    startCursor
    endCursor
    hasNextPage
    __typename
        
      }
    __typename
        
      }
    appearsIn
    
    ...on Human {
      
    homePlanet
    height
    mass
      starships {
        
    id
    name
    length
    coordinates
    __typename
        
      }
    }
    ...on Droid {
      
    primaryFunction
    }
  }
}
",
  "reviews" => "
query Reviews 
  (
      $episode:Episode!
  )
{
  reviews
  (
      episode:$episode
  )
 
   {
    
    episode
    stars
    commentary
    __typename
    
  }
}
",
  "search" => "
query Search 
  (
      $text:String
  )
{
  search
  (
      text:$text
  )
 
   {
    
    
    ...on Human {
      
    id
    name
    homePlanet
    height
    mass
      friends {
        
    id
    name
      friends {
        
    id
    name
    appearsIn
        
    ...on Human {
      
    homePlanet
    height
    mass
    }
    ...on Droid {
      
    primaryFunction
    }
      }
      friendsConnection {
        
    totalCount
    __typename
        
      }
    appearsIn
        
    ...on Human {
      
    homePlanet
    height
    mass
      starships {
        
    id
    name
    length
    coordinates
    __typename
        
      }
    }
    ...on Droid {
      
    primaryFunction
    }
      }
      friendsConnection {
        
    totalCount
      edges {
        
    cursor
    __typename
        
      }
      friends {
        
    id
    name
    appearsIn
        
    ...on Human {
      
    homePlanet
    height
    mass
    }
    ...on Droid {
      
    primaryFunction
    }
      }
      pageInfo {
        
    startCursor
    endCursor
    hasNextPage
    __typename
        
      }
    __typename
        
      }
    appearsIn
      starships {
        
    id
    name
    length
    coordinates
    __typename
        
      }
    }
    ...on Droid {
      
    id
    name
      friends {
        
    id
    name
      friends {
        
    id
    name
    appearsIn
        
    ...on Human {
      
    homePlanet
    height
    mass
    }
    ...on Droid {
      
    primaryFunction
    }
      }
      friendsConnection {
        
    totalCount
    __typename
        
      }
    appearsIn
        
    ...on Human {
      
    homePlanet
    height
    mass
      starships {
        
    id
    name
    length
    coordinates
    __typename
        
      }
    }
    ...on Droid {
      
    primaryFunction
    }
      }
      friendsConnection {
        
    totalCount
      edges {
        
    cursor
    __typename
        
      }
      friends {
        
    id
    name
    appearsIn
        
    ...on Human {
      
    homePlanet
    height
    mass
    }
    ...on Droid {
      
    primaryFunction
    }
      }
      pageInfo {
        
    startCursor
    endCursor
    hasNextPage
    __typename
        
      }
    __typename
        
      }
    appearsIn
    primaryFunction
    }
    ...on Starship {
      
    id
    name
    length
    coordinates
    }
  }
}
",
  "character" => "
query Character 
  (
      $id:ID!
  )
{
  character
  (
      id:$id
  )
 
   {
    
    id
    name
      friends {
        
    id
    name
      friends {
        
    id
    name
    appearsIn
        
    ...on Human {
      
    homePlanet
    height
    mass
    }
    ...on Droid {
      
    primaryFunction
    }
      }
      friendsConnection {
        
    totalCount
    __typename
        
      }
    appearsIn
        
    ...on Human {
      
    homePlanet
    height
    mass
      starships {
        
    id
    name
    length
    coordinates
    __typename
        
      }
    }
    ...on Droid {
      
    primaryFunction
    }
      }
      friendsConnection {
        
    totalCount
      edges {
        
    cursor
    __typename
        
      }
      friends {
        
    id
    name
    appearsIn
        
    ...on Human {
      
    homePlanet
    height
    mass
    }
    ...on Droid {
      
    primaryFunction
    }
      }
      pageInfo {
        
    startCursor
    endCursor
    hasNextPage
    __typename
        
      }
    __typename
        
      }
    appearsIn
    
    ...on Human {
      
    homePlanet
    height
    mass
      starships {
        
    id
    name
    length
    coordinates
    __typename
        
      }
    }
    ...on Droid {
      
    primaryFunction
    }
  }
}
",
  "droid" => "
query Droid 
  (
      $id:ID!
  )
{
  droid
  (
      id:$id
  )
 
   {
    
    id
    name
      friends {
        
    id
    name
      friends {
        
    id
    name
    appearsIn
        
    ...on Human {
      
    homePlanet
    height
    mass
    }
    ...on Droid {
      
    primaryFunction
    }
      }
      friendsConnection {
        
    totalCount
    __typename
        
      }
    appearsIn
        
    ...on Human {
      
    homePlanet
    height
    mass
      starships {
        
    id
    name
    length
    coordinates
    __typename
        
      }
    }
    ...on Droid {
      
    primaryFunction
    }
      }
      friendsConnection {
        
    totalCount
      edges {
        
    cursor
    __typename
        
      }
      friends {
        
    id
    name
    appearsIn
        
    ...on Human {
      
    homePlanet
    height
    mass
    }
    ...on Droid {
      
    primaryFunction
    }
      }
      pageInfo {
        
    startCursor
    endCursor
    hasNextPage
    __typename
        
      }
    __typename
        
      }
    appearsIn
    primaryFunction
    __typename
    
  }
}
",
  "human" => "
query Human 
  (
      $id:ID!
  )
{
  human
  (
      id:$id
  )
 
   {
    
    id
    name
    homePlanet
    height
    mass
      friends {
        
    id
    name
      friends {
        
    id
    name
    appearsIn
        
    ...on Human {
      
    homePlanet
    height
    mass
    }
    ...on Droid {
      
    primaryFunction
    }
      }
      friendsConnection {
        
    totalCount
    __typename
        
      }
    appearsIn
        
    ...on Human {
      
    homePlanet
    height
    mass
      starships {
        
    id
    name
    length
    coordinates
    __typename
        
      }
    }
    ...on Droid {
      
    primaryFunction
    }
      }
      friendsConnection {
        
    totalCount
      edges {
        
    cursor
    __typename
        
      }
      friends {
        
    id
    name
    appearsIn
        
    ...on Human {
      
    homePlanet
    height
    mass
    }
    ...on Droid {
      
    primaryFunction
    }
      }
      pageInfo {
        
    startCursor
    endCursor
    hasNextPage
    __typename
        
      }
    __typename
        
      }
    appearsIn
      starships {
        
    id
    name
    length
    coordinates
    __typename
        
      }
    __typename
    
  }
}
",
  "starship" => "
query Starship 
  (
      $id:ID!
  )
{
  starship
  (
      id:$id
  )
 
   {
    
    id
    name
    length
    coordinates
    __typename
    
  }
}
",
}
`;

exports[`GraphQL documents generation tests for introspection schema input should generate GraphQL documents for schema 2`] = `
Map {
  "createReview" => "
mutation CreateReview 
  (
      $episode:Episode,
      $review:ReviewInput!
  )
{
  createReview
  (
      episode:$episode,
      review:$review
  )
 
   {
    
    episode
    stars
    commentary
    __typename
    
  }
}
",
}
`;

exports[`GraphQL documents generation tests for introspection schema input should generate GraphQL documents for schema 3`] = `
Map {
  "reviewAdded" => "
subscription ReviewAdded 
  (
      $episode:Episode
  )
{
  reviewAdded
  (
      episode:$episode
  )
 
   {
    
    episode
    stars
    commentary
    __typename
    
  }
}
",
}
`;

exports[`end 2 end tests to test if the case style is retained for type names should generate statements 1`] = `
Map {
  "getUPPERCASE" => "
query GetUPPERCASE 
  (
      $id:ID!
  )
{
  getUPPERCASE
  (
      id:$id
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "listUPPERCASEs" => "
query ListUPPERCASEs 
  (
      $filter:ModelUPPERCASEFilterInput,
      $limit:Int,
      $nextToken:String
  )
{
  listUPPERCASEs
  (
      filter:$filter,
      limit:$limit,
      nextToken:$nextToken
  )
 
   {
    
      items {
        
    id
    owner
    createdAt
    updatedAt
    __typename
        
      }
    nextToken
    __typename
    
  }
}
",
  "getLowercase" => "
query GetLowercase 
  (
      $id:ID!
  )
{
  getLowercase
  (
      id:$id
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "listLowercases" => "
query ListLowercases 
  (
      $filter:ModellowercaseFilterInput,
      $limit:Int,
      $nextToken:String
  )
{
  listLowercases
  (
      filter:$filter,
      limit:$limit,
      nextToken:$nextToken
  )
 
   {
    
      items {
        
    id
    owner
    createdAt
    updatedAt
    __typename
        
      }
    nextToken
    __typename
    
  }
}
",
  "getCamelCase" => "
query GetCamelCase 
  (
      $id:ID!
  )
{
  getCamelCase
  (
      id:$id
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "listCamelCases" => "
query ListCamelCases 
  (
      $filter:ModelcamelCaseFilterInput,
      $limit:Int,
      $nextToken:String
  )
{
  listCamelCases
  (
      filter:$filter,
      limit:$limit,
      nextToken:$nextToken
  )
 
   {
    
      items {
        
    id
    owner
    createdAt
    updatedAt
    __typename
        
      }
    nextToken
    __typename
    
  }
}
",
  "getPascalCase" => "
query GetPascalCase 
  (
      $id:ID!
  )
{
  getPascalCase
  (
      id:$id
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "listPascalCases" => "
query ListPascalCases 
  (
      $filter:ModelPascalCaseFilterInput,
      $limit:Int,
      $nextToken:String
  )
{
  listPascalCases
  (
      filter:$filter,
      limit:$limit,
      nextToken:$nextToken
  )
 
   {
    
      items {
        
    id
    owner
    createdAt
    updatedAt
    __typename
        
      }
    nextToken
    __typename
    
  }
}
",
  "getSnake_case" => "
query GetSnake_case 
  (
      $id:ID!
  )
{
  getSnake_case
  (
      id:$id
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "listSnake_cases" => "
query ListSnake_cases 
  (
      $filter:Modelsnake_caseFilterInput,
      $limit:Int,
      $nextToken:String
  )
{
  listSnake_cases
  (
      filter:$filter,
      limit:$limit,
      nextToken:$nextToken
  )
 
   {
    
      items {
        
    id
    owner
    createdAt
    updatedAt
    __typename
        
      }
    nextToken
    __typename
    
  }
}
",
  "getUPPER_SNAKE_CASE" => "
query GetUPPER_SNAKE_CASE 
  (
      $id:ID!
  )
{
  getUPPER_SNAKE_CASE
  (
      id:$id
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "listUPPER_SNAKE_CASEs" => "
query ListUPPER_SNAKE_CASEs 
  (
      $filter:ModelUPPER_SNAKE_CASEFilterInput,
      $limit:Int,
      $nextToken:String
  )
{
  listUPPER_SNAKE_CASEs
  (
      filter:$filter,
      limit:$limit,
      nextToken:$nextToken
  )
 
   {
    
      items {
        
    id
    owner
    createdAt
    updatedAt
    __typename
        
      }
    nextToken
    __typename
    
  }
}
",
  "queryByOwner" => "
query QueryByOwner 
  (
      $owner:String,
      $sortDirection:ModelSortDirection,
      $filter:ModelUPPERCASEFilterInput,
      $limit:Int,
      $nextToken:String
  )
{
  queryByOwner
  (
      owner:$owner,
      sortDirection:$sortDirection,
      filter:$filter,
      limit:$limit,
      nextToken:$nextToken
  )
 
   {
    
      items {
        
    id
    owner
    createdAt
    updatedAt
    __typename
        
      }
    nextToken
    __typename
    
  }
}
",
}
`;

exports[`end 2 end tests to test if the case style is retained for type names should generate statements 2`] = `
Map {
  "createUPPERCASE" => "
mutation CreateUPPERCASE 
  (
      $input:CreateUPPERCASEInput!,
      $condition:ModelUPPERCASEConditionInput
  )
{
  createUPPERCASE
  (
      input:$input,
      condition:$condition
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "updateUPPERCASE" => "
mutation UpdateUPPERCASE 
  (
      $input:UpdateUPPERCASEInput!,
      $condition:ModelUPPERCASEConditionInput
  )
{
  updateUPPERCASE
  (
      input:$input,
      condition:$condition
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "deleteUPPERCASE" => "
mutation DeleteUPPERCASE 
  (
      $input:DeleteUPPERCASEInput!,
      $condition:ModelUPPERCASEConditionInput
  )
{
  deleteUPPERCASE
  (
      input:$input,
      condition:$condition
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "createLowercase" => "
mutation CreateLowercase 
  (
      $input:CreateLowercaseInput!,
      $condition:ModellowercaseConditionInput
  )
{
  createLowercase
  (
      input:$input,
      condition:$condition
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "updateLowercase" => "
mutation UpdateLowercase 
  (
      $input:UpdateLowercaseInput!,
      $condition:ModellowercaseConditionInput
  )
{
  updateLowercase
  (
      input:$input,
      condition:$condition
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "deleteLowercase" => "
mutation DeleteLowercase 
  (
      $input:DeleteLowercaseInput!,
      $condition:ModellowercaseConditionInput
  )
{
  deleteLowercase
  (
      input:$input,
      condition:$condition
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "createCamelCase" => "
mutation CreateCamelCase 
  (
      $input:CreateCamelCaseInput!,
      $condition:ModelcamelCaseConditionInput
  )
{
  createCamelCase
  (
      input:$input,
      condition:$condition
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "updateCamelCase" => "
mutation UpdateCamelCase 
  (
      $input:UpdateCamelCaseInput!,
      $condition:ModelcamelCaseConditionInput
  )
{
  updateCamelCase
  (
      input:$input,
      condition:$condition
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "deleteCamelCase" => "
mutation DeleteCamelCase 
  (
      $input:DeleteCamelCaseInput!,
      $condition:ModelcamelCaseConditionInput
  )
{
  deleteCamelCase
  (
      input:$input,
      condition:$condition
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "createPascalCase" => "
mutation CreatePascalCase 
  (
      $input:CreatePascalCaseInput!,
      $condition:ModelPascalCaseConditionInput
  )
{
  createPascalCase
  (
      input:$input,
      condition:$condition
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "updatePascalCase" => "
mutation UpdatePascalCase 
  (
      $input:UpdatePascalCaseInput!,
      $condition:ModelPascalCaseConditionInput
  )
{
  updatePascalCase
  (
      input:$input,
      condition:$condition
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "deletePascalCase" => "
mutation DeletePascalCase 
  (
      $input:DeletePascalCaseInput!,
      $condition:ModelPascalCaseConditionInput
  )
{
  deletePascalCase
  (
      input:$input,
      condition:$condition
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "createSnake_case" => "
mutation CreateSnake_case 
  (
      $input:CreateSnake_caseInput!,
      $condition:Modelsnake_caseConditionInput
  )
{
  createSnake_case
  (
      input:$input,
      condition:$condition
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "updateSnake_case" => "
mutation UpdateSnake_case 
  (
      $input:UpdateSnake_caseInput!,
      $condition:Modelsnake_caseConditionInput
  )
{
  updateSnake_case
  (
      input:$input,
      condition:$condition
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "deleteSnake_case" => "
mutation DeleteSnake_case 
  (
      $input:DeleteSnake_caseInput!,
      $condition:Modelsnake_caseConditionInput
  )
{
  deleteSnake_case
  (
      input:$input,
      condition:$condition
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "createUPPER_SNAKE_CASE" => "
mutation CreateUPPER_SNAKE_CASE 
  (
      $input:CreateUPPER_SNAKE_CASEInput!,
      $condition:ModelUPPER_SNAKE_CASEConditionInput
  )
{
  createUPPER_SNAKE_CASE
  (
      input:$input,
      condition:$condition
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "updateUPPER_SNAKE_CASE" => "
mutation UpdateUPPER_SNAKE_CASE 
  (
      $input:UpdateUPPER_SNAKE_CASEInput!,
      $condition:ModelUPPER_SNAKE_CASEConditionInput
  )
{
  updateUPPER_SNAKE_CASE
  (
      input:$input,
      condition:$condition
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "deleteUPPER_SNAKE_CASE" => "
mutation DeleteUPPER_SNAKE_CASE 
  (
      $input:DeleteUPPER_SNAKE_CASEInput!,
      $condition:ModelUPPER_SNAKE_CASEConditionInput
  )
{
  deleteUPPER_SNAKE_CASE
  (
      input:$input,
      condition:$condition
  )
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
}
`;

exports[`end 2 end tests to test if the case style is retained for type names should generate statements 3`] = `
Map {
  "onCreateUPPERCASE" => "
subscription OnCreateUPPERCASE 
{
  onCreateUPPERCASE
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "onUpdateUPPERCASE" => "
subscription OnUpdateUPPERCASE 
{
  onUpdateUPPERCASE
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "onDeleteUPPERCASE" => "
subscription OnDeleteUPPERCASE 
{
  onDeleteUPPERCASE
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "onCreateLowercase" => "
subscription OnCreateLowercase 
{
  onCreateLowercase
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "onUpdateLowercase" => "
subscription OnUpdateLowercase 
{
  onUpdateLowercase
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "onDeleteLowercase" => "
subscription OnDeleteLowercase 
{
  onDeleteLowercase
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "onCreateCamelCase" => "
subscription OnCreateCamelCase 
{
  onCreateCamelCase
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "onUpdateCamelCase" => "
subscription OnUpdateCamelCase 
{
  onUpdateCamelCase
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "onDeleteCamelCase" => "
subscription OnDeleteCamelCase 
{
  onDeleteCamelCase
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "onCreatePascalCase" => "
subscription OnCreatePascalCase 
{
  onCreatePascalCase
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "onUpdatePascalCase" => "
subscription OnUpdatePascalCase 
{
  onUpdatePascalCase
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "onDeletePascalCase" => "
subscription OnDeletePascalCase 
{
  onDeletePascalCase
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "onCreateSnake_case" => "
subscription OnCreateSnake_case 
{
  onCreateSnake_case
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "onUpdateSnake_case" => "
subscription OnUpdateSnake_case 
{
  onUpdateSnake_case
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "onDeleteSnake_case" => "
subscription OnDeleteSnake_case 
{
  onDeleteSnake_case
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "onCreateUPPER_SNAKE_CASE" => "
subscription OnCreateUPPER_SNAKE_CASE 
{
  onCreateUPPER_SNAKE_CASE
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "onUpdateUPPER_SNAKE_CASE" => "
subscription OnUpdateUPPER_SNAKE_CASE 
{
  onUpdateUPPER_SNAKE_CASE
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
  "onDeleteUPPER_SNAKE_CASE" => "
subscription OnDeleteUPPER_SNAKE_CASE 
{
  onDeleteUPPER_SNAKE_CASE
 
   {
    
    id
    owner
    createdAt
    updatedAt
    __typename
    
  }
}
",
}
`;
